tests/webapp/api/test_performance_data_api.py

```diff
@@ -0,0 +1,115 @@
+import datetime
+import time
+
+import pytest
+from django.core.urlresolvers import reverse
+from rest_framework.test import APIClient
+
+from treeherder.model.models import (MachinePlatform,
+                                     Option,
+                                     OptionCollection)
+from treeherder.perf.models import (PerformanceDatum,
+                                    PerformanceFramework,
+                                    PerformanceSignature)
+
+
+@pytest.fixture
+def summary_perf_signature(test_repository, test_perf_signature):
+
+    framework = PerformanceFramework.objects.get(
+        name='test_talos')
+    option = Option.objects.create(name='opt')
+    option_collection = OptionCollection.objects.create(
+        option_collection_hash='my_option_hash',
+        option=option)
+    platform = MachinePlatform.objects.create(
+        os_name='win',
+        platform='win7',
+        architecture='x86',
+        active_status='active')
+
+    # summary performance signature don't have test value
+    signature = PerformanceSignature.objects.create(
+        repository=test_repository,
+        signature_hash=(40*'s'),
+        framework=framework,
+        platform=platform,
+        option_collection=option_collection,
+        suite='mysuite'
+    )
+    signature.subtests.add(test_perf_signature)
+    return signature
+
+
+def test_no_summary_performance_date(webapp, test_repository, test_perf_signature,
+                                     jm):
+
+    INTERVAL = 30
+    now = time.time()
+    for (t, v) in zip([i for i in range(INTERVAL)],
```

Comment:
There's no need to create tons of PerformanceDatum objects here. Just one should suffice.